@here/olp-sdk-dataservice-read/lib/StatisticsRequest.ts

```diff
@@ -51,22 +53,42 @@ export class StatisticsRequest {
         return this.dataLevel;
     }
 
-    public withCatalogHrn(hrn: string) {
-        this.catalogHrn = hrn;
+    /**
+     * Setter for the provided hrn
+     * @param hrn
+     * @returns this to have ability to chain methods
+     */
+    public withCatalogHrn(hrn: HRN): StatisticsRequest {
+        this.catalogHrn = hrn.toString();
         return this;
     }
 
-    public withLayerId(layerId: string) {
+    /**
+     * Setter for the provided layerId
+     * @param layerId
+     * @returns this to have ability to chain methods
+     */
+    public withLayerId(layerId: string): StatisticsRequest {
         this.layerId = layerId;
         return this;
     }
 
-    public withTypemap(type: CoverageDataType) {
-        this.typemap = type;
+    /**
+     * Setter for the provided coverageDataType
+     * @param coverageDataType
+     * @returns this to have ability to chain methods
+     */
+    public withTypemap(coverageDataType: CoverageDataType): StatisticsRequest {
+        this.typemap = coverageDataType;
         return this;
     }
 
-    public withDataLevel(dataLevel: string) {
+    /**
+     * Setter for the provided dataLevel
+     * @param dataLevel
+     * @returns this to have ability to chain methods
+     */
+    public withDataLevel(dataLevel: string): StatisticsRequest {
```

Comment:
Please add more info for setters, like: "Tile level. Currently, should be equal or less than level 11", Required or not, etc ...

Reply:
Done